src/bounds.jl

```diff
@@ -1,22 +1,104 @@
-    m.u_var_tight = [m.u_var_orig,fill(Inf, m.num_var_linear_lifted_mip+m.num_var_nonlinear_lifted_mip);]
+    m.l_var_tight = [m.l_var_orig, fill(-Inf, m.num_var_linear_mip+m.num_var_nonlinear_mip);]
+    m.u_var_tight = [m.u_var_orig, fill(Inf, m.num_var_linear_mip+m.num_var_nonlinear_mip);]
     for i in 1:m.num_var_orig
         if m.var_type_orig[i] == :Bin
             m.l_var_tight[i] = 0.0
             m.u_var_tight[i] = 1.0
+        elseif m.var_type_orig[i] == :Int
+            m.l_var_tight[i] = floor(m.l_var_tight[i])
+            m.u_var_tight[i] = ceil(m.u_var_tight[i])
         end
     end
 
     return
 end
 
+"""
+    init_disc(m::PODNonlinearModel)
+
+This function initialize the dynamic discretization used for any bounding models. By default, it takes (.l_var_orig, .u_var_orig) as the base information. User is allowed to use alternative bounds for initializing the discretization dictionary.
+The output is a dictionary with MathProgBase variable indices keys attached to the :PODNonlinearModel.discretization.
+"""
+function init_disc(m::PODNonlinearModel)
+
+    for var in 1:(m.num_var_orig+m.num_var_linear_mip+m.num_var_nonlinear_mip)
+        if m.var_type[var] in [:Bin, :Cont]
+            lb = m.l_var_tight[var]
+            ub = m.u_var_tight[var]
+            m.discretization[var] = [lb, ub]
+        elseif m.var_type[var] in [:Int]
+            m.int_enable ? lb = floor(m.l_var_tight[var]) - 0.5 : lb = floor(m.l_var_tight[var])
+            m.int_enable ? ub = ceil(m.u_var_tight[var]) + 0.5 : ub = floor(m.u_var_tight[var])
+            m.discretization[var] = [lb, ub]
+        else
+            error("[EXCEPTION] Unexpected variable type when initializing discretization dictionary.")
+        end
+    end
+
+    return
+end
+
+"""
+
+    to_discretization(m::PODNonlinearModel, lbs::Vector{Float64}, ubs::Vector{Float64})
+
+Utility functions to convert bounds vectors to Dictionary based structures that is more suitable for
+partition operations.
+
+"""
+function to_discretization(m::PODNonlinearModel, lbs::Vector{Float64}, ubs::Vector{Float64})
+
+    @assert length(lbs) == length(ubs)
+    var_discretization = Dict()
+    for var in 1:m.num_var_orig
+        lb = lbs[var]
+        ub = ubs[var]
+        var_discretization[var] = [lb, ub]
+    end
+
+    total_var_cnt = m.num_var_orig+m.num_var_linear_mip+m.num_var_nonlinear_mip
+    orig_var_cnt = m.num_var_orig
+
+    if length(lbs) == total_var_cnt
+        for var in (1+orig_var_cnt):total_var_cnt
+            lb = lbs[var]
+            ub = ubs[var]
+            var_discretization[var] = [lb, ub]
+        end
+    else
+        for var in (1+orig_var_cnt):total_var_cnt
+            lb = -Inf
+            ub = Inf
+            var_discretization[var] = [lb, ub]
+        end
+    end
+
+    return var_discretization
+end
+
+"""
+    flatten_discretization(discretization::Dict)
```

Comment:
Again, I don't understand what this function does

Reply:
This is a function that we used during BT, which cancels all the partitions information (of a given partition dictionary) and only keeps the original bounds (from l_var_tight and u_var_tight).